network/proxy.go

```diff
@@ -553,9 +553,14 @@ func (pr *Proxy) PassThroughToClient(conn *ConnWrapper, stack *Stack) *gerr.Gate
 		span.AddEvent("Plugin(s) modified the response")
 	}
 
-	// Send the response to the client.
-	errVerdict := pr.sendTrafficToClient(conn.Conn(), response, received)
-	span.AddEvent("Sent traffic to client")
+	var errVerdict *gerr.GatewayDError
+	if received > 0 {
+		// Send the response to the client.
+		errVerdict = pr.sendTrafficToClient(conn.Conn(), response, received)
+		span.AddEvent("Sent traffic to client")
+	} else {
+		span.AddEvent("No data to send to client")
+	}
```

Comment:
I initially assumed that each request to PostgreSQL would have a response, but I was proven wrong by the [`COPY` command](https://www.postgresql.org/docs/current/protocol-flow.html#PROTOCOL-COPY). When the client sends a `COPY` command to the server (postgres), the server will reply with a `CopyInResponse` command, and provided that no errors happen anywhere in our systems, the client starts sending `CopyData` requests, which might be more than one request depending on the chunk size. After each `CopyData` request nothing is sent back by the server (no notification of receiveing the requests), until the client sends a `CopyDone` message at the end of all the `CopyData` requests. Only then the server responds with a `CommandComplete` and a `ReadyForQuery` response. Thus, the client might send multiple requests without receiving a response after each one. And the `ReadyForQuery` response from the server causes the client to start sending the next request.

Ref: https://www.postgresql.org/docs/current/protocol-message-formats.html